sklearn/tree/_utils.pyx

```diff
@@ -361,14 +362,8 @@ cdef class WeightedPQueue:
         # Resize if capacity not sufficient
         if array_ptr >= self.capacity:
             self.capacity *= 2
-            array = <WeightedPQueueRecord*> realloc(self.array_,
-                                                    self.capacity *
-                                                    sizeof(WeightedPQueueRecord))
-
-            if array == NULL:
-                # no free; __dealloc__ handles that
-                return -1
-            self.array_ = array
```

Comment:
not quite relevant to the issue at hand, but do you think these same fixes should be applied for the other data structures in utils.pyx?